precise/iaf_psc_exp_ps_time_reversal.cpp

```diff
@@ -0,0 +1,720 @@
+/*
+ *  iaf_psc_exp_ps_time_reversal.cpp
+ *
+ *  This file is part of NEST.
+ *
+ *  Copyright (C) 2004 The NEST Initiative
+ *
+ *  NEST is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  NEST is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+#include "iaf_psc_exp_ps_time_reversal.h"
+
+#include "exceptions.h"
+#include "dict.h"
+#include "integerdatum.h"
+#include "doubledatum.h"
+#include "dictutils.h"
+#include "universal_data_logger_impl.h"
+#include "arraydatum.h"
+
+#include <limits>
+
+/* ---------------------------------------------------------------- 
+ * Recordables map
+ * ---------------------------------------------------------------- */
+
+nest::RecordablesMap<nest::iaf_psc_exp_ps_time_reversal> nest::iaf_psc_exp_ps_time_reversal::recordablesMap_;
+
+namespace nest
+{
+  // Override the create() method with one call to RecordablesMap::insert_() 
+  // for each quantity to be recorded.
+  template <>
+  void RecordablesMap<iaf_psc_exp_ps_time_reversal>::create()
+  {
+    // use standard names whereever you can for consistency!
+    insert_(names::V_m, & iaf_psc_exp_ps_time_reversal::get_V_m_);
+    insert_(names::I_syn, & iaf_psc_exp_ps_time_reversal::get_I_syn_);
+    insert_(names::y1_ex, & iaf_psc_exp_ps_time_reversal::get_y1_ex_);
+    insert_(names::y1_in, &iaf_psc_exp_ps_time_reversal::get_y1_in_);
```

Comment:
If I understand the code correctly, these are the excitatory and inhibitory synaptic current. Please use the pre-defined names `I_syn_ex` and `I_syn_in`, respectively.